### DISCUSSION TITLE
[Client caching] Prevent loading server data which is (actually) non-stale on device

### DISCUSSION THREAD

**Discussion by seyfeb** [ZiggyCreatures/FusionCache#88] [Q&A]
Hey there!

Maybe someone can help me understand what's the best way to handle the following scenario:

Assume you are using the cache on a client, e.g., a mobile device to (also) store big (i.e., expensive to load) data. You would never want to load the data if it has not changed on the server and the same version is still available from the cache.

The way I see to handle this is: Set an **infinitely long lifetime** for the cache entries. When requesting an item, manually check the server each time (with an If-Modified-Since header). If it has been **updated on the server**, **update the cache** and somehow inform the requesting object about the update.

Is there a better approach, e.g., a combination of some of FusionCache features, that is targeting such a use case and would make the "manual" part superfluous?

**Comment by jodydonetti** [comment]
Hi @seyfeb ,
first of all this is a very interesting scenario, and something I've been thinking about from some time ([conditional requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests) in general, with all of their varitions of headers).

Your approach seems overall fine, but I'm thinking about some details to be sure it will perform well.

Lately I've been quite busy with my daily job (just switched to a new one!) so please give me some time to come back to you with a proof of concept about how to solve this scenario well.

Will update as soon as possible.

**Reply by seyfeb** [reply]
Thanks for the effort, I'm not expecting immediate support in your free time ;) But at least, i hope, any outcome might help not only me but also others facing the same problems!

I appreciate you taking the time pondering all these things and bringing discussions and this library forward! Good luck with your new job!
